docs/source/tutorials/spark.rst

```diff
@@ -0,0 +1,191 @@
+           df = df.drop('c')
+           self.write(df)
+   
+   
+   @luisy.requires(TaskA)
+   @luisy.deltatable_output(schema='my_schema', catalog='my_catalog', table_name='final')
+   class TaskB(SparkTask):
+   
+       def run(self):
+           df = self.input().read()
+           df = df.withColumn('f', 2*df.a)
+           self.write(df)
+   
+   @luisy.requires(TaskB)
+   @luisy.final
+   @luisy.pickle_output
+   class TaskC(SparkTask):
+       def run(self):
+           df = self.input().read()
+           self.write(df)
+
+
+Here, :code:`TaskA` and :code:`TaskB` read and write their data from
+and to delta tables and process them with spark. :code:`TaskC`,
+however, persists its output into a pickle file, which requires
+:py:mod:`luisy` to serialize all the data to a
+:py:mod:`pandas.DataFrame` beforehand.
+
+Running a pipeline
+------------------
+
+When the pipeline should be executed within an active python session,
+running the pipeline can be done as follows:
+
+.. code-block:: python
+
+   from luisy.cli import build
+
+   build(TaskC(), cloud_mode=True)
+   
+In this case, the :py:class:`pyspark.SparkContext()` is automatically
+propagated to :py:mod:`luisy` from the active session. Alternatively,
+if a special spark context has to be used, the spark context need to
+be attached to the :py:class:`~luisy.config.Config` first as follows:
+
+.. code-block:: python
+
+
+   Config().set_param('spark', some_predefined_spark_context) 
+
+
+For instance, this could be a spark instance created via spark
+connect:
+
+
+.. code-block:: python
+
+   spark = SparkSession.builder.remote("sc://my-cluster:15002").getOrCreate()
+   Config().set_param('spark', spark) 
+
+Be aware that all :py:class:`~luisy.targets.LocalTarget` point to
+locations on the system of the python session where :py:mod:`luisy`
+runs in.
+
+.. _databricks:
+
+Using databricks
+----------------
+
+A convinient way to interact with pyspark clusters is by using the
+databricks abstraction through a databricks notebook. Its
+also possible to connect from a local session using
+:py:mod:`databricks_connect` (see :ref:`databricks-connect`).
+
+.. note::
+
+   When using :py:mod:`luisy` in a databricks cluster, additional
+   charges are generated for the user. The amount of expenses depends
+   among others on the cloud provider and the cluster configuration.
+   :py:mod:`luisy` has no influences on the generated costs and we
+   recommend to monitor cloud costs closely.
+
+
+.. note::
+   The tasks itself cannot be implemented within the notebook and need
+   to be implemented in a standalone python package or module.  Only
+   execution can be done via a databricks notebook.
+
+
+Initial configuration
+~~~~~~~~~~~~~~~~~~~~~
+
+Using :py:mod:`luisy` within a databricks cluster, the databricks file
+system (:code:`dbfs`) can be used as local file system allowing to run
+the pipeline completely in
+the cloud, even non-:py:class:`~luisy.tasks.base.SparkTask`.
```

Comment:
even **for** non Spark tasks ?

Reply:
Exactly, even `luisy.tasks.base.Task` objects are executed (technically) "in the cloud", meaning within the python session in which the notebook runs. May be an overstatement, I just wanted to highlight that nothing is executed locally.

Reply:
sorry, I wanted to comment about the grammar / snytax of your sentence .... Is there a **for** missing ?

Reply:
Hups, sorry! Yes, you are completely right. Just fixed it.